extensions/sql-database-projects/src/models/project.ts

```diff
@@ -306,80 +306,63 @@ export class Project implements ISqlProject {
 
 	/**
 	 * Adds a folder to the project, and saves the project file
+	 *
 	 * @param relativeFolderPath Relative path of the folder
-	 * @param doNotThrowOnDuplicate
-	 *	Flag that indicates whether duplicate entries should be ignored or throw an error. If flag is set to `true` and
-	 *	item already exists in the project file, then existing entry will be returned.
 	 */
-	public async addFolderItem(relativeFolderPath: string, doNotThrowOnDuplicate?: boolean): Promise<FileProjectEntry> {
-		const absoluteFolderPath = path.join(this.projectFolderPath, relativeFolderPath);
-		const normalizedRelativeFolderPath = utils.convertSlashesForSqlProj(relativeFolderPath);
+	public async addFolderItem(relativeFolderPath: string): Promise<FileProjectEntry> {
+		const folderEntry = await this.ensureFolderItems(relativeFolderPath);
```

Comment:
I refactored the code to reuse the `addFolderItem` logic from the `addScriptItem`, but there is small caveat. In case someone is trying to add the project folder itself or anything outside it - we throw an error (`constants.outsideFolderPath`). For the `addScriptItem`, though, someone may add a file in the root of the project folder, thus this will be called with the project folder itself and throw an error. Instead of complicating the code in the `addScriptItem` and special-case files in the project root, I just extracted the logic in a separate helper, which return `undefined` instead of throwing an error and used that in the `addScriptItem`.